client/boot/app.js

```diff
@@ -20,13 +17,15 @@ import { configureReduxStore, locales, setupMiddlewares, utils } from './common'
 import createReduxStoreFromPersistedInitialState from 'state/initial-state';
 import detectHistoryNavigation from 'lib/detect-history-navigation';
 import userFactory from 'lib/user';
+import * as wordpressProject from './project/wordpress-com';
 
 const debug = debugFactory( 'calypso' );
+const projects = { 'wordpress-com': wordpressProject };
 
 const boot = currentUser => {
 	debug( "Starting Calypso. Let's do this." );
 
-	const project = require( `./project/${ PROJECT_NAME }` );
+	const project = projects[ PROJECT_NAME ];
```

Comment:
It is a different implementation. It will always load the project which is s valid use case for WordPress.com, but it prevents us from making it customizable. The idea here was to load only code for a one project specified in config file. Webpack optimizes everything behind the scenes.